### PR TITLE
Some changes for the 0.8 list of changes.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: docs/080-breaking-changes.rst [hunk 1]
```diff
@@ -86,12 +89,16 @@ New Restrictions
...
 
 Interface Changes
 =================
 
-* Changed output of ``--combined-json``. JSON fields ``abi``, ``devdoc``, ``userdoc`` and ``storage-layout`` are sub-objects now. Before 0.8.0 they used to be serialised as strings.
+* The output of ``--combined-json`` has changed: JSON fields ``abi``, ``devdoc``, ``userdoc`` and
+  ``storage-layout`` are sub-objects now. Before 0.8.0 they used to be serialised as strings.
+
+* The "legacy AST" has been removed (``--ast-json`` on the commandline interface and ``legacyAST`` for standard-json).
+  Use the "cempact AST" (``--ast-compact--json`` resp. ``AST``) as replacement.
```
Comment (axic) [inline docs/080-breaking-changes.rst]:
```suggestion
  Use the "compact AST" (``--ast-compact--json`` resp. ``AST``) as replacement.
```

Comment (hrkrshnn) [inline docs/080-breaking-changes.rst]:
```suggestion
  Use the "compact AST" (``--ast-compact--json`` resp. ``AST``) as the replacement.
```

File: docs/080-breaking-changes.rst [hunk 2]
```diff
@@ -86,12 +89,16 @@ New Restrictions
   This will make it safe to assume that the underlying type in the ABI is always ``uint8``.
 
 * Declarations with the name ``this``, ``super`` and ``_`` are disallowed, with the exception of
-  public functions and events.
+  public functions and events to make it possible to interface with languages other than Solidity.
 
 * Remove support for the ``\b``, ``\f``, and ``\v`` escape sequences in code.
   They can still be inserted via hexadecimal escapes, e.g. ``\x08``, ``\x0c``, and ``\x0b``, respectively.
 
 Interface Changes
```
Comment (axic) [inline docs/080-breaking-changes.rst:111]: Mention the error reporter?

Comment (chriseth) [reply docs/080-breaking-changes.rst:111]: Oh, that is not even in the changelog, is it?

**Comment by chriseth** [comment]
Added suggestions and mentioned the error reporter (it actually is in the changelog).

File: docs/080-breaking-changes.rst [hunk 3]
```diff
@@ -86,12 +89,18 @@ New Restrictions
   This will make it safe to assume that the underlying type in the ABI is always ``uint8``.
 
 * Declarations with the name ``this``, ``super`` and ``_`` are disallowed, with the exception of
-  public functions and events.
+  public functions and events to make it possible to interface with languages other than Solidity.
```
Comment (axic) [inline docs/080-breaking-changes.rst]: I don't fully understand the relation here.

File: docs/080-breaking-changes.rst [hunk 4]
```diff
@@ -86,12 +89,18 @@ New Restrictions
...
 =================
 
-* Changed output of ``--combined-json``. JSON fields ``abi``, ``devdoc``, ``userdoc`` and ``storage-layout`` are sub-objects now. Before 0.8.0 they used to be serialised as strings.
+* The output of ``--combined-json`` has changed: JSON fields ``abi``, ``devdoc``, ``userdoc`` and
+  ``storage-layout`` are sub-objects now. Before 0.8.0 they used to be serialised as strings.
+
+* The "legacy AST" has been removed (``--ast-json`` on the commandline interface and ``legacyAST`` for standard-json).
+  Use the "compact AST" (``--ast-compact--json`` resp. ``AST``) as replacement.
+
+* The legacy error reporter has been removed.
```
Comment (axic) [inline docs/080-breaking-changes.rst]: Explain what this is? E.g. the commandline option

**Comment by chriseth** [comment]
Better now?

File: docs/080-breaking-changes.rst [hunk 5]
```diff
@@ -86,12 +89,19 @@ New Restrictions
...
 =================
 
-* Changed output of ``--combined-json``. JSON fields ``abi``, ``devdoc``, ``userdoc`` and ``storage-layout`` are sub-objects now. Before 0.8.0 they used to be serialised as strings.
+* The output of ``--combined-json`` has changed: JSON fields ``abi``, ``devdoc``, ``userdoc`` and
+  ``storage-layout`` are sub-objects now. Before 0.8.0 they used to be serialised as strings.
+
+* The "legacy AST" has been removed (``--ast-json`` on the commandline interface and ``legacyAST`` for standard-json).
+  Use the "compact AST" (``--ast-compact--json`` resp. ``AST``) as replacement.
+
+* The old error reporter (``solc --old-reporter``) has been removed.
```
Comment (axic) [inline docs/080-breaking-changes.rst]: The other lines didn't refer to `solc`:
```suggestion
* The old error reporter (``--old-reporter``) has been removed.
```

File: docs/080-breaking-changes.rst [hunk 6]
```diff
@@ -86,12 +89,19 @@ New Restrictions
...
   They can still be inserted via hexadecimal escapes, e.g. ``\x08``, ``\x0c``, and ``\x0b``, respectively.
 
 Interface Changes
 =================
 
-* Changed output of ``--combined-json``. JSON fields ``abi``, ``devdoc``, ``userdoc`` and ``storage-layout`` are sub-objects now. Before 0.8.0 they used to be serialised as strings.
+* The output of ``--combined-json`` has changed: JSON fields ``abi``, ``devdoc``, ``userdoc`` and
+  ``storage-layout`` are sub-objects now. Before 0.8.0 they used to be serialised as strings.
+
+* The "legacy AST" has been removed (``--ast-json`` on the commandline interface and ``legacyAST`` for standard-json).
```
Comment (axic) [inline docs/080-breaking-changes.rst]: Are we really using "standard-json" as a term?  think we've used different style.

**Review by axic — APPROVED** [review]
Mostly nitpicks, but looks good.

File: docs/080-breaking-changes.rst (hunk 6 above)
Comment (chriseth) [reply docs/080-breaking-changes.rst]: We sometimes call it 'standard JSON interface', sometimes 'JSON-input-output interface".

Comment (axic) [reply docs/080-breaking-changes.rst]: How about we introduce a new term (was thinking about this problem for a while, just never had an idea): "JSON Compiler Interface" or "JCI" for short. And avoid using the input-output and standard, etc.

**Pull request merged**